UpcomingMovies.Core/Services/HttpClientService.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace UpcomingMovies.Core.Services
+{
+    public class HttpClientService : IHttpClientService
+    {
+        HttpClient _client;
+
+        public HttpClientService()
+        {
+            _client = new HttpClient();
+            _client.Timeout = TimeSpan.FromSeconds(10);
+        }
+
+        public Task<HttpResponseMessage> GetAsync(string url)
+        {
+            return _client.GetAsync(url);
```

Comment:
What if you make this method `async` and write `return await ...`? Does it make any difference?

Reply:
There is a small difference actually, by returning the Task instead of awaiting it, it changes the exception behavior of the method, as it won't throw the exception inside the method which starts the task but in the method which awaits it. It will also save the compiler from generating an extra async state machine.